src/main/java/eu/xenit/gradle/docker/compose/DockerComposeConventionImpl.java

```diff
@@ -108,23 +118,33 @@ public void fromProject(Project project) {
                 composeTask.doFirst(CONFIGURE_COMPOSE_ACTION_NAME, new Action<Task>() { // No lambda -> see #96
                     @Override
                     public void execute(Task t) {
-                        createSetComposeEnvironmentFromPathAction().execute(dockerBuildImage);
+                        if(environmentVariable != null) {
```

Comment:
Handling `null` here creates the additional case when users call `fromProject(null, ...)`, which was not allowed before.

I prefer not having a valid value of `null` for the `environmentVariable` parameter.

To accomplish this without having to duplicate code to the new methods that you introduced, you can make this function `private void fromProject0(String environmentVariable, Project project)` and introduce a new `fromProject(String environmentVariable, Project project)` function that will call this private function.
You will then check `environmentVariable` to be non-null with `Objects.requireNonNull(environmentVariable)`.